File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
from typing import List

class Solution:
    def maxProfit(self, l1: List[int]) -> int:
        l2 = []
        min1=l1[0]
        max1=0
        max2=0
        for i in l1[1:]:
            if i<min1:
                min1=i
               
            else:
                max1=max(max1,i-min1)
        return max1+max2

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int maxprofit=0;
        int l=0,r=1;
        while(r<prices.size()){
            if(maxprofit<prices[r]-prices[l]){
                maxprofit=prices[r]-prices[l];
            }
            if(prices[l]>prices[r]){
                l=r;
               
            }
    	
            r++;
        }
        return  maxprofit;
    }
};